flang/lib/Lower/CallInterface.cpp

```diff
@@ -461,17 +465,46 @@ class Fortran::lower::CallInterfaceImpl {
     }
   }
 
+  // Some dummy argument attributes prevent it from being used with implicit
+  // interfaces but that does not imply that a box is needed for those. For
+  // instance, scalar intrinsic that are OPTIONAL require an explicit interface,
+  // but they can still be passed by reference.
+  bool dummyRequiresBox(
```

Comment:
I'm a little confused by the comment vs. the code.

Specifically, I think this code does what it says: it is testing that the argument _must_ be passed in a box. However, the comment concerns when an argument in an implied interface might _not_ be passed in a box.

Reply:
Right, it was explaining the rational for this function, but that's not that useful. Replaced by "Define when an explicit argument must be passed in a fir.box."